cpp/src/messaging_adapter.cpp

```diff
@@ -116,7 +119,29 @@ void on_delivery(messaging_handler& handler, pn_event_t* event) {
     link_context& lctx = link_context::get(lnk);
     Tracing& ot = Tracing::getTracing();
 
-    if (pn_link_is_receiver(lnk)) {
+    if (pn_terminus_get_type(pn_link_remote_target(lnk)) == PN_COORDINATOR) {
+        if (pn_delivery_updated(dlv)) {
+            tracker t(make_wrapper<tracker>(dlv));
+            ot.on_settled_span(t);
+            switch (pn_delivery_remote_state(dlv)) {
+            case PN_ACCEPTED:
+                handler.on_tracker_accept(t);
+                break;
+            case PN_REJECTED:
+                handler.on_tracker_reject(t);
+                break;
+            case PN_RELEASED:
+            case PN_MODIFIED:
+                handler.on_tracker_release(t);
+                break;
+            }
+            if (t.settled()) {
+                handler.on_tracker_settle(t);
+                if (lctx.auto_settle)
+                    t.settle();
+            }
+        }
+    } else if (pn_link_is_receiver(lnk)) {
```

Comment:
I don't understand why the handling of messages on a transaction coordinator link should be different from any other link.